cli/command/service/opts.go

```diff
@@ -434,6 +436,11 @@ func newServiceOptions() *serviceOptions {
 func (opts *serviceOptions) ToService() (swarm.ServiceSpec, error) {
 	var service swarm.ServiceSpec
 
+	envVariables, err := runconfigopts.ReadKVStrings(opts.envFile.GetAll(), opts.env.GetAll())
```

Comment:
My thought behind this was that in services, a change in order can result in the service being re-deployed. Not sure if that's important in this case, but worth checking.